pkg/rec/rec_test.go

```diff
@@ -165,60 +176,69 @@ func TestRec(t *testing.T) {
 
 func TestNormalization(t *testing.T) {
 	tt := []struct {
-		in  string
-		out string
+		in        string
+		path, tag string
 	}{
 		{
-			in:  "a.b.c",
-			out: "a.b.c",
+			in:   "a.b.c",
+			path: "a.b.c",
 		},
 		{
-			in:  "a.b.c.",
-			out: "a.b.c",
+			in:   "a.b.c.",
+			path: "a.b.c",
 		},
 		{
-			in:  "abc.abc.abc",
-			out: "abc.abc.abc",
+			in:   "abc.abc.abc",
+			path: "abc.abc.abc",
 		},
 		{
-			in:  ".abc.abc.abc",
-			out: "abc.abc.abc",
+			in:   ".abc.abc.abc",
+			path: "abc.abc.abc",
 		},
 		{
-			in:  "...abc.abc.abc..",
-			out: "abc.abc.abc",
+			in:   "...abc.abc.abc..",
+			path: "abc.abc.abc",
 		},
 		{
-			in:  "abc..ab.abc",
-			out: "abc.ab.abc",
+			in:   "abc..ab.abc",
+			path: "abc.ab.abc",
 		},
 		{
-			in:  "abc..def..jkl..xyz",
-			out: "abc.def.jkl.xyz",
+			in:   "abc..def..jkl..xyz",
+			path: "abc.def.jkl.xyz",
 		},
 		{
-			in:  "ab&c",
-			out: "ab_c",
+			in:   "ab&c",
+			path: "ab_c",
 		},
 		{
-			in:  "ab   cd.a  b. zkl",
-			out: "ab___cd.a__b._zkl",
+			in:   "ab   cd.a  b. zkl",
+			path: "ab___cd.a__b._zkl",
 		},
 		{
-			in:  "ab^%+=.cdef.jk&",
-			out: "ab____.cdef.jk_",
+			in:   "ab^%+=.cdef.jk&",
+			path: "ab____.cdef.jk_",
+		},
+		{
+			in:   "ab^%+=.cdef.jk&;tag1=val1",
```

Comment:
Could you please add a couple more tests. It would be nice if they contained edge cases.